src/main/java/com/flowingcode/vaadin/addons/datetimerangepicker/ui/DateTimeRangePicker.java

```diff
@@ -0,0 +1,607 @@
+        this.startTime.setValue(minDate.isAfter(LocalTime.NOON) ? minDate : LocalTime.NOON);
+        this.endTime.setValue(maxDate.isBefore(LocalTime.NOON) ? maxDate : LocalTime.NOON);
+        this.endTime.setValue(maxDate);
+        revalidate();
+      }
+      this.startTime.setReadOnly(checked);
+      this.endTime.setReadOnly(checked);
+    });
+
+    allTimeChip.onPress(checked -> {
+      if (checked) {
+        LocalTime minDate = this.startTime.getMin() != null ? this.startTime.getMin() : LocalTime.MIN;
+        LocalTime maxDate = this.endTime.getMax() != null ? this.startTime.getMax() : LocalTime.MAX;
+        this.startTime.setValue(minDate);
+        this.endTime.setValue(maxDate);
+        revalidate();
+      }
+      this.startTime.setReadOnly(checked);
+      this.endTime.setReadOnly(checked);
+    });
+
+    this.timeDivider = new SpanLine();
+
+    HorizontalLayout headerLayout = new HorizontalLayout();
+    headerLayout.addClassNames(AlignItems.CENTER, Position.RELATIVE, JustifyContent.BETWEEN);
+
+    this.timeCircle = new Circle();
+
+    headerLayout.add(this.timeCircle, timesTitle, timeChipGroup);
+
+    HorizontalLayout selectorLayout = new HorizontalLayout();
+    selectorLayout.addClassNames(Gap.SMALL);
+    selectorLayout.add(this.startTime, this.timeDivider, this.endTime);
+
+    layout.add(headerLayout, selectorLayout);
+
+    return layout;
+  }
+
+  // Fires a revalidation when data changes
+  private void revalidate() {
+    getElement().executeJs("this.dispatchEvent(new Event('change'))");
+  }
+
+  // Custom field
+  @Override
+  protected DateTimeRange generateModelValue() {
+    return new DateTimeRange(
+        startDate.getValue(),
+        endDate.getValue(),
+        startTime.getValue(),
+        endTime.getValue(),
+        weekDays.getValue()
+    );
+  }
+
+  @Override
+  protected void setPresentationValue(DateTimeRange dateTimeRange) {
+
+    startDate.setValue(dateTimeRange.getStartDate());
+    endDate.setValue(dateTimeRange.getEndDate());
+    startTime.setValue(dateTimeRange.getStartTime());
+    endTime.setValue(dateTimeRange.getEndTime());
+    weekDays.setValue(dateTimeRange.getWeekDays());
+
+    daysDivider.setText(formatPeriod(dateTimeRange.getDaysSpan()));
+    Duration timeDuration = dateTimeRange.getDayDuration();
+    timeDivider.setText(formatDuration(timeDuration));
+  }
+
+  @Override
+  public Validator<DateTimeRange> getDefaultValidator() {
+    return new DateTimeRangePickerValidator(this);
+  }
+
+  // api
+
+  /**
+   * Changes this component visibility state
+   */
+  public void setVisible(boolean visible) {
+    super.setVisible(visible);
+    this.setDatesVisible(visible);
+    this.setDaysVisible(visible);
+    this.setTimesVisible(visible);
+  }
+
+  /**
+   * Changes the component read only state
+   */
+  public void setReadOnly(boolean readOnly) {
+    super.setReadOnly(readOnly);
```

Comment:
I get the following error with Eclipse
```
Cannot directly invoke the abstract method setReadOnly(boolean) for the type HasValue<AbstractField.ComponentValueChangeEvent<CustomField<DateTimeRange>,DateTimeRange>,DateTimeRange>
```

```suggestion
    getElement().setProperty("readonly", readOnly);
```

Reply:
Done. I use Intellij and it works, but apparently Eclipse forbids it.

![image](https://github.com/user-attachments/assets/eed64138-9954-4b06-ba47-83d10b624aba)

Reply:
IntelliJ uses javac while Eclipse has its own compiler (which sometimes don't agree on their interpretation of the Java Language spec)